### PR TITLE
initial fastlane structures

### PR DESCRIPTION
**Pull request by IzzySoft**
As discussed, here comes your starer-package for Fastlane. Be welcome to use my [Fastlane Cheat Sheet](https://gitlab.com/-/snippets/1895688) to guide you with building on this (e.g. add per-release changelogs or additional locales). If you have any questions, just ask :smile:

### CODE REVIEW

**Comment by Myzel394** [comment]
What a service :O @IzzySoft thank you very much!

**Comment by IzzySoft** [comment]
Once merged and that NonFreeDep solved, I'd say you're ready to apply at F-Droid.org for listing :smiley:

**Comment by Myzel394** [comment]
> Once merged and that NonFreeDep solved, I'd say you're ready to apply at F-Droid.org for listing 😃

Wait now I'm confused :O I thought "IzzyOnDroid" was like its own repo of apps (in contrast to the official F-Droid.org)

**Comment by Myzel394** [comment]
Oh wait I remember now why I needed GMS. I think it was required for getting the location. I can see if I can remove it, but if I remember correctly, it was required as otherwise the accuracy is pretty low.

**Comment by IzzySoft** [comment]
> I thought "IzzyOnDroid" was like its own repo of apps (in contrast to the official F-Droid.org)

It is (though rather "in complementing" instead of "in contrast" :wink:). But does that keep you from being listed there? No. Often, IoD serves as kind of "staging" (find out how close you are to comply with F-Droid's stronger criteria), then get fully compliant and move on – or don't and stay – or do both, get compliant, move on, AND stay (Star Trek's transporter wrongly configured: instead of moving, just performing a copy :see_no_evil:).

TL;DR: F-Droid.org has the higher prestige. Feel encouraged to aim that – but don't feel "forced to".

Disclosure: I'm also one of the F-Droid.org maintainers :wink: 

> Oh wait I remember now why I needed GMS. I think it was required for getting the location.

Ugh, really? Then I wonder how my devices do it. They all run Google-free. And I use only apps from F-Droid (well, except for TiBu and my good old RSS reader). In the field relevant here: Tower Collector, OsmAnd, Organic Maps, Street Complete, Every Door – they all work fine. Accuracy, depending on signal strength etc, down to 10m or less. So what do I wrong? None of those apps use GMS (or they wouldn't be on F-Droid).

> I can see if I can remove it, but if I remember correctly, it was required as otherwise the accuracy is pretty low.

Give it a try. You can also use a build flavor for that (e.g. `foss` for the "pure one" and `nonfree` for the other). Further, let's see what my [F-Droid snippet](https://gitlab.com/-/snippets/1896503) has on that topic… Ah, how fitting: [Flutter and Geolocation](https://gitlab.com/-/snippets/1896503#flutter-and-geolocation). And what do we [have here](https://github.com/Myzel394/locus/blob/master/pubspec.yaml#L49)? Seems to fit somehow, so something for you to read, maybe? Not sure if that section would need another update meanwhile, but it should definitely give you some good pointers I guess.

Oh, and thanks for the merge! Will now update the configuration on my end, so with future releases, Fastlane will be checked as well – and changes pulled over.

**Comment by IzzySoft** [comment]
> The only problem so far was that their documentation was not really easy to grasp.

It goes straight for the "supreme discipline" without mentioning the easier variants, yeah. See e.g. [here](https://android.izzysoft.de/articles/named/fdroid-intro-2#rfp) for the minimal set. Next higher level would be forking fdroiddata, creating your own branch, adding your `*.yml` file, and open a merge request – then let the CI do the testing (i.e. no need for a local "F-Droid build environment"). IMHO the "full variant" only makes sense if you either want to run your own repo and compile all kind of apps – or if you want to become a core contributor to the F-Droid team and help out with fixing builds. But as pointed out, that's **my** humble opinion, not necessarily that of everyone on the team (especially not the "official stance").

> Figured out I needed to actually go outside to receive GPS

Haha… yeah, depends on the device. My SHIFT6mq often gets a fix inside as well, my FP2 hardly finds more than 2 satellites. If you don't have any alternative set up to get NetworkLocation being used (I e.g. use microG), it takes a little longer. But I think it's worth it – at least that way Google isn't involved. I don't want to share my location (or other personal/private data) with them.

> I will no longer need to ping you, do I? Everything should run from now on automatically, shouldn't it?

For your app listing with my repo? Right, auto-pilot is engaged. If you have questions, I'm still here. And if you forgot to tell me about some extra "segment" configures with your Fastlane, there's a job running here once a month which would tell me :wink: I've pretty much everything automated here that can be automated. Still am able to trigger things manually (like, pulling a fresh copy of Fastlane without a new release being tagged).
